libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/helpers/logfile/LogFileHelpers.kt

```diff
@@ -0,0 +1,42 @@
+package org.wordpress.android.util.helpers.logfile
+
+import android.content.Context
+import java.io.File
+
+private const val LOG_FILE_DIRECTORY = "logs"
+
+/**
+ * A collection of helpers for Log Files.
+ */
+class LogFileProvider(private val logFileDirectoryPath: String) : LogFileProviderInterface {
+    /**
+     * Provides a {@link java.io.File} directory in which to store log files.
+     *
+     * If the directory doesn't already exist, it will be created.
+     */
+    override fun getLogFileDirectory(): File {
```

Comment:
This is super unimportant, but as we also talked about it on the screenshare session, I prefer to avoid doing multiple things in a function when I can avoid it. In this case avoiding it may not be the best strategy, so in these cases, I do something like this:

```
override fun getLogFileDirectory(createDirectoryIfNecessary: Boolean = true): File? {
```

In this case making this return a nullable has a cascading effect and I don't think we should do it, but I thought I'd mention this strategy for future reference. I think this type of approach makes the code more self-documenting.

_P.S: I wish we could specify a return type or just the nullability depending on a parameter._

Reply:
We don't have to do a nullable return here, we could just use a `check` to validate that the directory can be created.

The issue we hit is that it seems we can't use a default parameter in an overriding method for an interface. And an interface can't specify default parameter values. 

So I think we'd have to rework quite a few things to make this work, unfortunately.

Reply:
The nullable return would have been necessary because if the `createDirectoryIfNecessary` in that hypothetical function was `false`, we'd have to return `null` when the directory doesn't exist, right?

Reply:
Ah, yeah I suppose so. That's a weird API IMHO because it lets you really paint yourself into a corner though 🤔 

There's probably a better way to handle it, but it's not coming to me offhand – I think we could leave this for a future change.

Reply:
Definitely agreed, and that's why in my initial comment I said this was mostly mentioned for a future reference. I don't think there is a good way to do this for this feature, at least not in its current state.